lib/class-wp-rest-widget-updater-controller.php

```diff
@@ -34,13 +34,30 @@ public function __construct() {
 	public function register_routes() {
 			register_rest_route(
 				$this->namespace,
-				// Regex representing a PHP class extracted from http://php.net/manual/en/language.oop5.basic.php.
-				'/' . $this->rest_base . '/(?P<identifier>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)/',
+				'/' . $this->rest_base . '/',
```

Comment:
Hi @noisysocks, I ended up applying another big change to this PR.
I noticed that if we call the existing ajax admin endpoint without any changes, I retrieve its edit form. So instead of reimplementing this logic for reference widgets, I'm just relying on the endpoint.

For the class widgets, we can not use the endpoint, because there is not a widget instance at all. I kept the rest endpoint but modified it as suggested to be clear of what resource we provide a widget-form.
It may be useful e.g: to build an alternative widget edit UI, we can use this endpoint to retrieve a widget edit form and we can use the server-side render endpoint to retrieve the widget frontend.
I could have made it an ajax endpoint but the code would be more complex, e.g: we get validation and permission checked by the rest API.

Regarding the POST vs GET, the problem with get is that URLs have a max character limit depending on the browser. A widget may have a text area and use lots of character so it seemed safer to use POST. Also, widgets may have legitimate reasons to have side effects e.g: file uploads. Using a get this widgets would break.

Reply:
> I could have made it an ajax endpoint but the code would be more complex, e.g: we get validation and permission checked by the rest API.

I still have a slight preference for using an AJAX endpoint instead of a REST API endpoint because of how implementation-specific this endpoint is compared to the rest (ha!) of the REST API. Happy to bite my tongue and have us iterate on this, though. I'm glad that the endpoint has been marked experimental 🙂

It might be good to loop in the REST API team at some point on this question.

> Regarding the POST vs GET, the problem with get is that URLs have a max character limit depending on the browser. A widget may have a text area and use lots of character so it seemed safer to use POST. Also, widgets may have legitimate reasons to have side effects e.g: file uploads. Using a get this widgets would break.

Makes sense, thanks 👍